examples/mcu-board-support/stm32h735g.rs

```diff
@@ -389,3 +398,15 @@ impl slint::platform::Platform for StmBackend {
         defmt::println!("{=str}", arguments.to_string());
     }
 }
+
+fn rng(buf: &mut [u8]) -> Result<(), getrandom::Error> {
+    cortex_m::interrupt::free(|cs| match RNG.borrow(cs).borrow_mut().as_mut() {
+        Some(rng) => rng.fill(buf).map_err(|e| match e {
```

Comment:
Is it wise to call rng::fill  in a no-interrupt context?  how big can the buffer be?
Is that the recommended way to do it?

Reply:
Since the RNG is accessed in polling mode, it seems wrong to me to wait for the data ready bit and read the number while interrupts are enabled.

We wouldn't want an interrupt handler to steal the numbers.

I can't see anything in chapter 39 that suggests that interrupts should be enabled in polling mode either. The source of randomness is not related for interrupts AFAICS - a separate circuit is described instead.

The only other option I could think of for this code would be to take out the rng from the cell (replace the None), fill the buffer, and put it back. But that also seems like overkill to me.

Reply:
Ok, i'm just worried we keep the mutex for too long and that we miss critical interrupts because of that.

Reply:
I agree, that's a risk. At the moment we only call this once on startup to seed the rng in the slide puzzle I think. And generally I think we should use those to avoid exactly that dependency on the crypto rng that's in the device.